example.dev-config.yml

```diff
@@ -33,3 +33,4 @@ pulp_source_dir: "/home/vagrant/devel/pulpcore"
 #pulp_use_system_wide_pkgs: true
 pulp_settings:
   secret_key: "unsafe_default"
+  content_origin: "http://{{ ansible_fqdn }}"
```

Comment:
I didn't expect these were necessary because the ansible-pulp installer was setting them?

Reply:
The ansible-pulp implementation, to avoid difficult inter-role complexity, is to set it in the examples vars files rather than either the pulp or pulp-webserver role. Pulplift does not use that file.

Reply:
I mean, we could just set it in the `pulp` role which writes out the file. This is a trivial change to make, and I am honestly leaning towards it. 

However the value (port 80 rather than 24816) is determined by the fact that the `pulp-webserver` role is getting run. And we do not have an easy way of determining that `pulp-webserver` is getting run during the `pulp` role, especially when roles are dynamically included rather than statically imported.

If we were to set it in `pulp-webserver` then dynamically including the other roles (which have a static dendency on the `pulp` role) would statically include the `pulp`  role individually, which would write out the settings.py (without the value) in the other roles. At the end, it would get overwritten with the value when `pulp-webserver` gets dynamically included.

Determining how to, and revamping the whole dependencies between roles is on my to-do list. Newer versions of Ansible enable better ways of doing this I think.

Reply:
Actually, setting it in the `pulp` role would be relatively easy but not trivial. Because then we would need more merging logic. In order to default child variables (`pulp_settings.content_origin`) of a dict (`pulp_settings`) that the user is likely to put in the vars file, we need merging logic.